aws/aws-sdk-java-v2
# Enable JDK9+ code and builds to produce Java8 compatible jar

## Motivation and Context

* Newly developed Java9 code written with module-info.java/.class should still produce jars that run compatibly on JRE8 as this SDK is still required to support it.
* Developers should need to explicitly attempt to do this. Github workflows using JDK9+ should run this too.
* The publishing maven profile will want to do this by default, however the profile must be added first as calling maven with `-P <profile>` with a profile that does not exist will cause failure.



## Modifications

* Added Maven profile that triggers an explicit java-9-compile with appropriate options and java-8-compile excluding module-info.java. Java9+ class files (excluding module-info.class) are over-written with Java8 class files so the resulting jar runs on JRE8 successfully. 
* New profile has enforcement that JDK version is at least 9
* module-path-tests adjusted to suppresses both java-9-compile and java-8-compile execution, preferring how it has already been configured.

## Testing

1. Added new test sub-module under core/<module> set up with module-info.java and main method
1. Verify build succeeds when JAVA_HOME and PATH points to JDK8 and JDK9+ (tested with 17)
1. Execute main method through JRE8 classpath successfully
1. Execute main method through JRE8 with module-path successfully
1. Downstream build of in-progress Java9 module in separate fork/branch.



## Screenshots (if appropriate)

## Types of changes


## Checklist



## License
